tests/models/test_cooktop_answers.py

```diff
@@ -0,0 +1,109 @@
+"""
+Test energy consumption profile and behaviour of the CooktopAnswers class.
+"""
+
+from pytest import approx, raises
+
+from app.constants import DAYS_IN_YEAR
+from app.models.user_answers import CooktopAnswers, YourHomeAnswers
+from app.services.configuration import (
+    get_default_cooktop_answers,
+    get_default_your_home_answers,
+)
+
+
+def test_invalid_cooktop_type():
+    """
+    Test that an invalid cooktop type raises a ValueError.
+    """
+    your_home = YourHomeAnswers(
+        people_in_house=3,
+        postcode="1234",
+        disconnect_gas=True,
+    )
+    cooktop_answers = CooktopAnswers(
+        cooktop="Piped gas",
+        alternative_cooktop="Piped gas",
+    )
+    setattr(cooktop_answers, "cooktop", "Invalid type")
+    setattr(cooktop_answers, "alternative_cooktop", "Invalid type")
+    with raises(ValueError, match="Unknown cooktop type: Invalid type"):
+        cooktop_answers.energy_usage_pattern(your_home)
+
+
+def test_cooking_energy_usage():
+    """
+    Test the energy usage pattern for cooking.
+    """
+    your_home = get_default_your_home_answers()
+    cooktop = get_default_cooktop_answers()
+
+    # Modeled energy use in kWh for each cooktop type. This is based
+    # on a linearized energy use model that preserves the average
+    # household energy use for cooking. (See 'Cooking' sheet of
+    # supporting workbook.)
+    expected_energy_use = {
+        "Electric induction": [159, 239, 319, 398, 478, 558],
+        "Piped gas": [412, 618, 824, 1030, 1236, 1442],
+        "Bottled gas": [412, 618, 824, 1030, 1236, 1442, 1648],
```

Comment:
The last value for the Bottled gas, should it be there? The excel only have the values for 6 people. and all the other options have 6 values only.

Reply:
Nice catch!